duckduckgo/macos-browser
# Quick fix: Playlist IconOverlay scroll bug

Task/Issue URL: https://app.asana.com/0/970019368605465/1203190689300587
Tech Design URL:
CC:

**Description**:
The IconOverlay is `position:absolute` at the end of the `</body>`. If it's positioned above a video in a _scrollable playlist_ and the user scrolls the playlist, the it would be "floating" outside of the video element.

Expecting this to be quite rare, so instead of setting up a _third_ IconOverlay to be added within any playlist, we simply:
1. Hide the IconOverlay on scroll
2. Don't show the IconOverlay on top of a video in a playlist that is fully or partially scrolled outside of the playlist


**Steps to test this PR**:
1. With IconOverlay's turned ON:
2. Make a query on YouTube: "[random artist] playlist"
3. Click the playlist to open the first video in it
4. Hover videos in the playlist, positioned to the right of the video player.
5. Make sure functionality works as described in **Description**
6. Make sure there's no regressions (slight change to `window.onresize`, but shouldn't have any effect)

**Testing checklist**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
